src/core/opamSystem.ml

```diff
@@ -647,7 +647,12 @@ let extract_job ~dir file =
     else if try not (Sys.is_directory dir) with Sys_error _ -> false then
       internal_error "Extracting the archive would overwrite %s." dir
     else
-    match files_all_not_dir tmp_dir with
+    let flist =
+      OpamStd.Op.(
+        files_all_not_dir tmp_dir |>
+        List.filter (not @* OpamStd.String.contains ~sub:"pax_global_header"))
+    in
+    match flist with
```

Comment:
From my understanding, we not only need to remove the `pax_global_header` dir, but also move the contents of the other directory beside it to the root ?

Reply:
To have the same behavior no need to move the contents. With `tar` <1.14, there is a just an `pax_global_header` extra file at archive extract root, which confuses opam because there is different handling of copy for single-directory and several-file archive content.